src/tooltip/widget.ts

```diff
@@ -114,8 +114,20 @@ class TooltipWidget extends Widget {
     }
     switch (event.type) {
     case 'keydown':
+      if (this.node.contains(event.target as HTMLElement)) {
+        if ((event as KeyboardEvent).keyCode === 27) { // Escape key
+          this.dispose();
+        }
+        return;
+      }
+      this.dispose();
+      break;
     case 'mousedown':
-      this._dismiss(event);
+      if (this.node.contains(event.target as HTMLElement)) {
+        this.activate();
```

Comment:
@blink1073 Turns out I was mistaken re: clicks automatically focusing the parent node, hence the manual call to `activate`, here.